### PR TITLE
add support for php8

### PR DESCRIPTION
**Pull request by digibeuk**
##### Summary
- [X] Appropriate documentation has been written (check if not applicable).

**Breaking changes** 🔥
- None

**Backwards compatible additions** ✨
- None

**Bugfixes/Changed internals** 🎈
- None

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
...Standard.ControlStructures.BlockControlStructureSpacing,SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing,SlevomatCodingStandard.ControlStructures.RequireNullCoalesceOperator,SlevomatCodingStandard.ControlStructures.RequireYodaComparison,SlevomatCodingStandard.Exceptions.ReferenceThrowableOnly,SlevomatCodingStandard.Functions.StaticClosure,SlevomatCodingStandard.Namespaces.AlphabeticallySortedUses,SlevomatCodingStandard.Namespaces.FullyQualifiedGlobalConstants,SlevomatCodingStandard.Namespaces.FullyQualifiedGlobalFunctions,SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly,SlevomatCodingStandard.Namespaces.UselessAlias,SlevomatCodingStandard.Namespaces.UseSpacing,SlevomatCodingStandard.Namespaces.UnusedUses,SlevomatCodingStandard.TypeHints.LongTypeHints,SlevomatCodingStandard.TypeHints.NullableTypeForNullDefaultValue,SlevomatCodingStandard.TypeHints.NullTypeHintOnLastPosition,SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing,SlevomatCodingStandard.TypeHints.UselessConstantTypeHint,Squiz.Commenting.DocCommentAlignment,Squiz.Functions.MultiLineFunctionDeclaration,Squiz.WhiteSpace.SuperfluousWhitespace --parallel=`nproc --all` -q --report=checkstyle src | cs2pr
```
Comment (jkiesebrink) [inline .github/workflows/ci.yml]: where did you get this sniffer configuration, I don't see it in the `Makefile`
```suggestion
        run: php vendor/bin/phpcs --parallel=`nproc --all` -q --report=checkstyle | cs2pr
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+name: Backend CI
```
Comment (jkiesebrink) [inline .github/workflows/ci.yml]: There is no other CI here, e.g. frontend?
```suggestion
name: CI
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      - 'app/schema/**.xml'
+      - 'composer.lock'
+      - 'phpcs.xml.dist'
+      - 'phpunit.xml.dist'
+      - 'psalm.xml'
+      - 'psalm-baseline.xml'
+
+jobs:
+  test:
+    name: Test
```
Comment (jkiesebrink) [inline .github/workflows/ci.yml]:
```suggestion
    name: Test (PHP ${{ matrix.php-version }})
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    paths:
+      - '**.php'
+      - '**.yml'
+      - '**.yaml'
+      - 'app/schema/**.xml'
+      - 'composer.lock'
+      - 'phpcs.xml.dist'
+      - 'phpunit.xml.dist'
+      - 'psalm.xml'
+      - 'psalm-baseline.xml'
```
Comment (jkiesebrink) [inline .github/workflows/ci.yml]: Why not just `on: push`, there is no psalm, yml/yaml files in this project?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Setup Environment
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version}}
+          coverage: none
+          extensions: bcmath, exif, gd, gmp, imagick, intl, mysqli, opcache, pcntl, pdo_mysql, redis, soap, sockets, tidy, zip
```
Comment (jkiesebrink) [inline .github/workflows/ci.yml]: Why? according to the `composer.json` there are not requirements for any extension

**Review by jkiesebrink — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 6]
```diff
@@ -1,14 +0,0 @@
-language: php
```
Comment (gertjuhh) [inline .travis.yml:1]: If you're going to remove Travis, the build status should also be removed from the README

File: .github/workflows/ci.yml (hunk 3 above)
Comment (gertjuhh) [reply .github/workflows/ci.yml]: I don't think you have access to the matrix at this point.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (gertjuhh) [reply .github/workflows/ci.yml]: No use running this on push to master, so wouldn't go with a simple `on: push`.
Do agree there are more paths listed then needed.
```suggestion
on:
  push:
    branches-ignore:
      - 'master'
    tags-ignore:
      - '**'
    paths:
      - '**.php'
      - 'composer.lock'
      - 'phpcs.xml.dist'
      - 'phpunit.xml.dist'
```

**Review by gertjuhh — CHANGES_REQUESTED** [review]

**Review by gertjuhh — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (jkiesebrink) [reply .github/workflows/ci.yml]: So this is not going to work either? https://github.com/MyOnlineStore/common-factories/blob/8e07275aacb360e94ff64248c476504a4a233498/.github/workflows/ci.yml#L7

Comment (gertjuhh) [reply .github/workflows/ci.yml]: Seems to be working :shrug:

Comment (jkiesebrink) [reply .github/workflows/ci.yml]: I don't quite understand the magic behind the title of the job name, in the monolith we also test with a matrix but the `test.name` is just "Test", yet the version is behind it https://github.com/MyOnlineStore/myonlinestore/blob/02fc69bc41a89ef05fad56ac68ee7f22bcad8c96/.github/workflows/backend-ci.yml#L22

Comment (gertjuhh) [reply .github/workflows/ci.yml]: There's a slight difference in outcome, `Test (7.2)` vs `Test( PHP 7.2)`
You're suggestion is a bit more explicit, but I don't have a strong preference on which way we go.
https://github.com/MyOnlineStore/common-factories/actions/runs/668325245
https://github.com/MyOnlineStore/myonlinestore/actions/runs/1097418223

**Review by jkiesebrink — APPROVED** [review]

**Pull request merged**